lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -763,10 +787,12 @@ LogicalResult UIntType::verify(function_ref<InFlightDiagnostic()> emitError,
 //===----------------------------------------------------------------------===//
 
 struct circt::firrtl::detail::BundleTypeStorage : mlir::TypeStorage {
-  using KeyTy = ArrayRef<BundleType::BundleElement>;
+  using ElementsType = ArrayRef<BundleType::BundleElement>;
+  using KeyTy = std::pair<ArrayRef<BundleType::BundleElement>, StringAttr>;
```

Comment:
Isn't the description string part of BundleElement?  What's the pair for?

Reply:
Wait, I see later the string is the bundle name the fields are part of.

Reply:
Moved descriptions out of this PR.